mods/timer-resolution-control.wh.cpp

```diff
@@ -215,25 +221,84 @@ BOOL Wh_ModInit(void)
     ULONG CurrentResolution;
     NTSTATUS status = ((NTSTATUS(WINAPI*)(PULONG, PULONG, PULONG))pNtQueryTimerResolution)(
         &MinimumResolution, &MaximumResolution, &CurrentResolution);
-    if (NT_SUCCESS(status)) {
-        Wh_Log(L"NtQueryTimerResolution: min=%f, max=%f, current=%f",
-            (double)MinimumResolution / 10000.0,
-            (double)MaximumResolution / 10000.0,
-            (double)CurrentResolution / 10000.0);
-        g_minimumResolution = MinimumResolution;
-        g_maximumResolution = MaximumResolution;
+    if (!NT_SUCCESS(status)) {
+        Wh_Log(L"NtQueryTimerResolution failed with status: 0x%X", status);
+        return FALSE;
     }
 
+    Wh_Log(L"NtQueryTimerResolution: min=%f, max=%f, current=%f",
+        (double)MinimumResolution / 10000.0,
+        (double)MaximumResolution / 10000.0,
+        (double)CurrentResolution / 10000.0);
+    g_minimumResolution = MinimumResolution;
+    g_maximumResolution = MaximumResolution;
+    g_lastDesiredResolution = CurrentResolution;    //TODO: During mod load there is a small race condition where the hook is not yet set, the g_lastDesiredResolution is already set, and then the program tries to set a new desired resolution which will not be saved into g_lastDesiredResolution.
+
     LoadSettings();
 
     Wh_SetFunctionHook((void*)pNtSetTimerResolution, (void*)NtSetTimerResolutionHook, (void**)&pOriginalNtSetTimerResolution);
 
     return TRUE;
 }
 
-void Wh_ModSettingsChanged(void)
+void EnforceLimits() 
```

Comment:
This function is incompatible with the `block` config. Which also has a small bug I just discovered: by setting `g_limitResolution = g_minimumResolution;`, setting the resolution isn't completely blocked, `g_minimumResolution` can still be set.